### PR TITLE
Kernel+lscpu: Expose/show size of L1 data/instruction, L2, and L3 CPU caches

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Kernel/GlobalProcessExposed.cpp [hunk 1]
```diff
@@ -639,6 +640,38 @@ class ProcFSCPUInformation final : public ProcFSGlobalInformation {
                 TRY(obj.add("type", info.type()));
                 TRY(obj.add("brand", info.brand_string()));
 
+                auto caches = TRY(obj.add_object("caches"));
+
+                if (info.l1_data_cache().has_value()) {
+                    auto l1_data = TRY(caches.add_object("l1_data"));
```
Comment (golosovsky) [inline Kernel/GlobalProcessExposed.cpp:646]: Hey, that's a really cool feature :)

Sorry for intervening, I've just watched your video and wanted to check out the PR :) 
How would you feel about extracting the boilerplate code which is duplicated here per each cache information category / type into a small helper method or a static func, which will do the boilerplate logic for you?

For example, by introducing that method / func (treat as pseudo-code please 😅):
```cpp
    template <typename T>
    ErrorOr<void> add_cache_info(JsonObjectSerializer<Builder> const& caches_list, KString const& new_cache_name, Optional<Cache> const& new_cache_details)
    {
        // create field for new cache
        auto new_cache = TRY(caches_list.add_object(new_cache_name));
        
        // populate new cache detauls
        TRY(new_cache.add("size", new_cache_details->size));
        TRY(new_cache.add("line_size", new_cache_details->line_size));
        TRY(new_cache.finish());
        return {};
    }

```

One would be able to rewrite the cache-info-retrieval logic into something like:
```cpp
auto caches = TRY(obj.add_object("caches"));

if (info.l1_data_cache().has_value())
    add_cache_info(caches, "l1_data", info.l1_data_cache());

if (info.l1_instruction_cache().has_value())
    add_cache_info(caches, "l1_instruction", info.l1_instruction_cache();

if (info.l2_cache().has_value())
    add_cache_info(caches, "l2", info.l2_cache();

if (info.l3_cache().has_value())
    add_cache_info(caches, "l3", info.l3_cache();

TRY(caches.finish());
```

Which just seems more readable to me, and will allow to easily change that piece of logic more easily :)

This logic can be also extracted into a separate `populate_cache_info` method – which will accept a non-const reference to the JSON builder and populate it with the retrieved cache information – to make `try_generate` even more readable.

Watcha think? I'd love to hear your opinion on that.

**Review by golosovsky — COMMENTED** [review]
Sorry for intervening, I've just watched your video and wanted to check out the PR :)

**Review by golosovsky — COMMENTED** [review]
minor comment update

File: Kernel/GlobalProcessExposed.cpp (hunk 1 above)
Comment (linusg) [reply Kernel/GlobalProcessExposed.cpp]: Sure, done :)

**Pull request merged**